Add some simple telephony and audio controls to MBS.

dthkao opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/google/mobly-bundled-snippets/14)
<!-- Reviewable:end -->

xpconanfan commented on src/main/java/com/google/android/mobly/snippet/bundled/AudioSnippet.java:
```diff
@@ -0,0 +1,59 @@
+    }
+  
+    @Rpc(description = "Sets the media stream volume.")
+    public void setMediaVolume(Integer value) {
```
We should have a prefix for all of these method in this snippet?

dthkao replied:
I'll leave this decision to you and Alex. I think if we start to go this route we might as well make separate MBS snippets to keep name spaces separate at the client (and permissions required separate).

xpconanfan replied:
After much thought, I'm ok with these names as they are now...

xpconanfan commented on src/main/java/com/google/android/mobly/snippet/bundled/AudioSnippet.java:
```diff
@@ -0,0 +1,59 @@
+  }
+
+    @Rpc(description = "Mute media stream.")
+    public void muteMedia() {
```
I'm not sure what the difference is between muteMedia and muteRing, perhaps a simple 'mute' makes more sense to normal users?

dthkao replied:
Anyone who has owned an android phone knows that there are multiple volume settings... I can add a mute that will do both both we'll eventually need separate ones (audio quality).

dthkao replied:
the separate streams in the api have existed since the beginning but I think exposing these in UI is relatively new. Added "mute" to mute both streams.

xpconanfan commented on src/main/java/com/google/android/mobly/snippet/bundled/TelephonySnippet.java:
```diff
@@ -0,0 +1,49 @@
+import com.google.android.mobly.snippet.Snippet;
+import com.google.android.mobly.snippet.rpc.Rpc;
+
+/** Snippet class exposing Android APIs in WifiManager. */
```
'in WifiManager'?

dthkao replied:
Fixed

xpconanfan commented on src/main/java/com/google/android/mobly/snippet/bundled/TelephonySnippet.java:
```diff
@@ -0,0 +1,49 @@
+/** Snippet class exposing Android APIs in WifiManager. */
+public class TelephonySnippet implements Snippet {
+
+    private final TelephonyManager telephonyManager;
```
Perhaps it's good to use telecome manager for higher API versions?
Also, can we have the basic calling APIs if we are adding this snippet?

dthkao replied:
Will add them later. They'd need considerably more logic and only these two are blocking right now.

xpconanfan commented on src/main/java/com/google/android/mobly/snippet/bundled/AudioSnippet.java:
```diff
@@ -0,0 +1,59 @@
+import android.telephony.TelephonyManager;
+import com.google.android.mobly.snippet.Snippet;
+import com.google.android.mobly.snippet.rpc.Rpc;
+import com.google.android.mobly.snippet.util.Log;
```
never used?

dthkao replied:
Fixed

xpconanfan commented:
Can you also add these (Requested by an internal customer):
1. `getAllCellInfo`
2. `GetSubscriberId`
3. [toggle tele data](http://stackoverflow.com/questions/29340150/android-l-5-x-turn-on-off-mobile-data-programmatically)

xpconanfan commented on src/main/java/com/google/android/mobly/snippet/bundled/TelephonySnippet.java:
```diff
@@ -0,0 +1,60 @@
+
+    @Rpc(description = "Returns all observed cell information from all radios on the device" +
+            " including the primary and neighboring cells.")
+    public int getAllCellInfo() {
```
The return type of `getAllCellInfo` should be `List<CellInfo>`?
Have you tested each Rpc you added?

dthkao replied:
no, i'll test when you're done adding rpcs to add.

dthkao commented:
Did not add a getter/setter for mobile data state since this would need a little more work. but added the mute rpc and tele get methods.

xpconanfan commented on src/main/java/com/google/android/mobly/snippet/bundled/TelephonySnippet.java:
```diff
@@ -0,0 +1,60 @@
+
+    @Rpc(description = "Returns the unique subscriber ID, for example, the IMSI for a GSM phone.")
+    public int getSubscriberId() {
+        return telephonyManager.getSubscriberId();
```
This seems to return String instead of int?

dthkao replied:
yep, whoops.

xpconanfan commented on src/main/java/com/google/android/mobly/snippet/bundled/AudioSnippet.java:
```diff
@@ -0,0 +1,73 @@
+    }
+
+    @Rpc(description = "Silences both the ringer and media audio.")
+    public void mute() {
```
Should this be called `muteAll`?

dthkao replied:
done, but potentially a misnomer since there are multiple streams we do not touch (system, dtmf, alarm, notification)

xpconanfan replied:
Should we call this `muteAudio` then?
Or make `muteAll` mute all of those streams?
Trying to avoid ambiguity in the method names.

dthkao replied:
done.

xpconanfan commented on src/main/java/com/google/android/mobly/snippet/bundled/AudioSnippet.java:
```diff
@@ -0,0 +1,73 @@
+    }
+
+    @Rpc(description = "Gets the ringer volume.")
+    public void getRingVolume() {
```
the name is `getXxx` yet it returns nothing?

dthkao replied:
fixed.

xpconanfan commented on src/main/java/com/google/android/mobly/snippet/bundled/AudioSnippet.java:
```diff
@@ -0,0 +1,73 @@
+    }
+  
+    @Rpc(description = "Gets the media stream volume.")
+    public void getMediaVolume() {
```
the name is `getXxx` yet it returns nothing?

dthkao replied:
fixed.

xpconanfan commented on src/main/java/com/google/android/mobly/snippet/bundled/AudioSnippet.java:
```diff
@@ -33,8 +33,8 @@ public AudioSnippet() {
     @Rpc(description = "Gets the media stream volume.")
-    public void getMediaVolume() {
-        audioManager.getStreamVolume(AudioManager.STREAM_MUSIC);
+    public int getMediaVolume() {
```
Should this be `getMusicVolume`?
What's the rationale behind using `Media`?

xpconanfan commented on src/main/java/com/google/android/mobly/snippet/bundled/TelephonySnippet.java:
```diff
@@ -39,13 +39,13 @@ public String getLine1Number() {
     @Rpc(description = "Returns all observed cell information from all radios on the device" +
             " including the primary and neighboring cells.")
-    public int getAllCellInfo() {
+    public List<CellInfo> getAllCellInfo() {
```
What's in CellInfo?
Does Gson do the job or do we need to add a specific method for it in JsonBuilder?

adorokhine commented:
Reviewed 1 of 2 files at r3, 1 of 1 files at r4, 1 of 1 files at r5.
Review status: all files reviewed at latest revision, 16 unresolved discussions.

---

*[src/main/java/com/google/android/mobly/snippet/bundled/AudioSnippet.java, line 28 at r5](https://reviewable.io:443/reviews/google/mobly-bundled-snippets/14#-Kcu4BYcggovOcdxY7Y7:-Kcu4BYd9Vdpu3DQPLGS:b-2hrsyl) ([raw file](https://github.com/google/mobly-bundled-snippets/blob/1e26712496c67c66a7871fb4e0ec1de197b741dc/src/main/java/com/google/android/mobly/snippet/bundled/AudioSnippet.java#L28)):*
> ```Java
> public class AudioSnippet implements Snippet {
> 
>     private final AudioManager audioManager;
> ```

should be named mAudioManager per the style guide

---

*[src/main/java/com/google/android/mobly/snippet/bundled/AudioSnippet.java, line 34 at r5](https://reviewable.io:443/reviews/google/mobly-bundled-snippets/14#-Kcu4H67VP_RAJTw8xwZ:-Kcu4H68oCIofxNNbCxu:b5dsros) ([raw file](https://github.com/google/mobly-bundled-snippets/blob/1e26712496c67c66a7871fb4e0ec1de197b741dc/src/main/java/com/google/android/mobly/snippet/bundled/AudioSnippet.java#L34)):*
> ```Java
>         this.audioManager = (AudioManager) context.getSystemService(Context.AUDIO_SERVICE);
>     }
>   
> ```

Indentation on empty line

---

*[src/main/java/com/google/android/mobly/snippet/bundled/AudioSnippet.java, line 36 at r5](https://reviewable.io:443/reviews/google/mobly-bundled-snippets/14#-Kcu4KymRVxMRspEJ9jW:-Kcu4KymRVxMRspEJ9jX:b-a4y2wm) ([raw file](https://github.com/google/mobly-bundled-snippets/blob/1e26712496c67c66a7871fb4e0ec1de197b741dc/src/main/java/com/google/android/mobly/snippet/bundled/AudioSnippet.java#L36)):*
> ```Java
>   
>     @Rpc(description = "Gets the media stream volume.")
>     public int getMediaVolume() {
> ```

Does the snippet lib handle the primitive int correctly?

---

*[src/main/java/com/google/android/mobly/snippet/bundled/AudioSnippet.java, line 36 at r5](https://reviewable.io:443/reviews/google/mobly-bundled-snippets/14#-Kcu4PnMeP31jXLewsYZ:-Kcu4PnMeP31jXLewsY_:by6ctzv) ([raw file](https://github.com/google/mobly-bundled-snippets/blob/1e26712496c67c66a7871fb4e0ec1de197b741dc/src/main/java/com/google/android/mobly/snippet/bundled/AudioSnippet.java#L36)):*
> ```Java
>   
>     @Rpc(description = "Gets the media stream volume.")
>     public int getMediaVolume() {
> ```

If it returns the STREAM_MUSIC volume, maybe it should be called getMusicVolume?

---

*[src/main/java/com/google/android/mobly/snippet/bundled/AudioSnippet.java, line 42 at r5](https://reviewable.io:443/reviews/google/mobly-bundled-snippets/14#-Kcu4UsfRqmQ_Embm8lT:-Kcu4UsfRqmQ_Embm8lU:b56q4hi) ([raw file](https://github.com/google/mobly-bundled-snippets/blob/1e26712496c67c66a7871fb4e0ec1de197b741dc/src/main/java/com/google/android/mobly/snippet/bundled/AudioSnippet.java#L42)):*
> ```Java
>     @Rpc(description = "Sets the media stream volume.")
>     public void setMediaVolume(Integer value) {
>         audioManager.setStreamVolume(AudioManager.STREAM_MUSIC, value, 0);
> ```

What does 0 mean? Callsite comments?

---

*[src/main/java/com/google/android/mobly/snippet/bundled/AudioSnippet.java, line 63 at r5](https://reviewable.io:443/reviews/google/mobly-bundled-snippets/14#-Kcu51bzxUKv9GEjtgdO:-Kcu51c-h-ndUUNQyoNu:bhoejdc) ([raw file](https://github.com/google/mobly-bundled-snippets/blob/1e26712496c67c66a7871fb4e0ec1de197b741dc/src/main/java/com/google/android/mobly/snippet/bundled/AudioSnippet.java#L63)):*
> ```Java
>         muteRing();  // STREAM_RING
>         audioManager.setStreamVolume(AudioManager.STREAM_SYSTEM, value, 0);
>         audioManager.setStreamVolume(AudioManager.STREAM_VOICE_CALL, value, 0);
> ```

Rather than having to maintain this list, can we just iterate over AudioSystem.getNumStreamTypes() and mute each one?

---

*[src/main/java/com/google/android/mobly/snippet/bundled/AudioSnippet.java, line 66 at r5](https://reviewable.io:443/reviews/google/mobly-bundled-snippets/14#-Kcu5Aeop_J2iRjZiJGT:-Kcu5Aeop_J2iRjZiJGU:bb8hn54) ([raw file](https://github.com/google/mobly-bundled-snippets/blob/1e26712496c67c66a7871fb4e0ec1de197b741dc/src/main/java/com/google/android/mobly/snippet/bundled/AudioSnippet.java#L66)):*
> ```Java
>     }
> 
>     @Rpc(description = "Puts the ringer volume at the lowest setting, but not set it DO NOT DISTURB.")
> ```

Does this mean vibration?

---

*[src/main/java/com/google/android/mobly/snippet/bundled/AudioSnippet.java, line 76 at r5](https://reviewable.io:443/reviews/google/mobly-bundled-snippets/14#-Kcu5D3moYEK-oSmx2ZQ:-Kcu5D3moYEK-oSmx2ZR:b-xu0n9a) ([raw file](https://github.com/google/mobly-bundled-snippets/blob/1e26712496c67c66a7871fb4e0ec1de197b741dc/src/main/java/com/google/android/mobly/snippet/bundled/AudioSnippet.java#L76)):*
> ```Java
>     }
> 
>     @Override
> ```

Alignment/indentation issues

---

*[src/main/java/com/google/android/mobly/snippet/bundled/TelephonySnippet.java, line 28 at r5](https://reviewable.io:443/reviews/google/mobly-bundled-snippets/14#-Kcu5Gk4trZnob3aTWjf:-Kcu5Gk4trZnob3aTWjg:bc4sdr1) ([raw file](https://github.com/google/mobly-bundled-snippets/blob/1e26712496c67c66a7871fb4e0ec1de197b741dc/src/main/java/com/google/android/mobly/snippet/bundled/TelephonySnippet.java#L28)):*
> ```Java
> public class TelephonySnippet implements Snippet {
> 
>     private final TelephonyManager telephonyManager;
> ```

mTelephonyManager

---

*[src/main/java/com/google/android/mobly/snippet/bundled/TelephonySnippet.java, line 47 at r5](https://reviewable.io:443/reviews/google/mobly-bundled-snippets/14#-Kcu5RHoRYm8qfIMLY68:-Kcu5RHoRYm8qfIMLY69:b-m3cmfq) ([raw file](https://github.com/google/mobly-bundled-snippets/blob/1e26712496c67c66a7871fb4e0ec1de197b741dc/src/main/java/com/google/android/mobly/snippet/bundled/TelephonySnippet.java#L47)):*
> ```Java
> 
>     @Rpc(description = "Returns all observed cell information from all radios on the device" +
>             " including the primary and neighboring cells.")
> ```

Should mention what type of object will be returned here.
Does the jsonrpcformatter already know how to handle CellInfo?

---

*[src/main/java/com/google/android/mobly/snippet/bundled/TelephonySnippet.java, line 52 at r5](https://reviewable.io:443/reviews/google/mobly-bundled-snippets/14#-Kcu5dNKZVITVuwGNC4i:-Kcu5dNM-c9-VZmmVD1r:b-cqo0br) ([raw file](https://github.com/google/mobly-bundled-snippets/blob/1e26712496c67c66a7871fb4e0ec1de197b741dc/src/main/java/com/google/android/mobly/snippet/bundled/TelephonySnippet.java#L52)):*
> ```Java
>     }
> 
>     @Rpc(description = "Gets the call state for the default subscription.")
> ```

Needs to mention what this int represents.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/google/mobly-bundled-snippets/14)*
<!-- Sent from Reviewable.io -->

dthkao commented:
*[src/main/java/com/google/android/mobly/snippet/bundled/AudioSnippet.java, line 36 at r3](https://reviewable.io:443/reviews/google/mobly-bundled-snippets/14#-Kcu3xDmqKYAsSQMyBnj-r3-36:-KcuuZ0VR5-llg0rxdjX:bf3eyf7) ([raw file](https://github.com/google/mobly-bundled-snippets/blob/61771354104ea110721b2d83a8c14d78f339e329/src/main/java/com/google/android/mobly/snippet/bundled/AudioSnippet.java#L36)):*
<details><summary><i>Previously, xpconanfan (Ang Li) wrote…</i></summary><blockquote>

Should this be `getMusicVolume`?
What's the rationale behind using `Media`?
</blockquote></details>

sl4a convention, and oddly enough it's labeled "media" in the volume settings menu. Happy to go either way on this.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/google/mobly-bundled-snippets/14)*
<!-- Sent from Reviewable.io -->

dthkao commented:
Review status: 1 of 3 files reviewed at latest revision, 18 unresolved discussions, some commit checks failed.

---

*[src/main/java/com/google/android/mobly/snippet/bundled/AudioSnippet.java, line 36 at r2](https://reviewable.io:443/reviews/google/mobly-bundled-snippets/14#-Kcu3xDmqKYAsSQMyBnj-r2-36:-KcvFnfJ1EufXbyiBG28:b-896fix) ([raw file](https://github.com/google/mobly-bundled-snippets/blob/7b89a773db6b163741a3161b2594366d562d8369/src/main/java/com/google/android/mobly/snippet/bundled/AudioSnippet.java#L36)):*
<details><summary><i>Previously, dthkao (David T.H. Kao) wrote…</i></summary><blockquote>

fixed.
</blockquote></details>

Done.

---

*[src/main/java/com/google/android/mobly/snippet/bundled/AudioSnippet.java, line 36 at r3](https://reviewable.io:443/reviews/google/mobly-bundled-snippets/14#-Kcu3xDmqKYAsSQMyBnj-r3-36:-KcvFokT7U105tOq37cr:b-5htgmd) ([raw file](https://github.com/google/mobly-bundled-snippets/blob/61771354104ea110721b2d83a8c14d78f339e329/src/main/java/com/google/android/mobly/snippet/bundled/AudioSnippet.java#L36)):*
<details><summary><i>Previously, dthkao (David T.H. Kao) wrote…</i></summary><blockquote>

sl4a convention, and oddly enough it's labeled "media" in the volume settings menu. Happy to go either way on this.
</blockquote></details>

Converted to music.

---

*[src/main/java/com/google/android/mobly/snippet/bundled/AudioSnippet.java, line 28 at r5](https://reviewable.io:443/reviews/google/mobly-bundled-snippets/14#-Kcu4BYcggovOcdxY7Y7:-KcuxkLcxSVlX7W7-lSI:b-896fix) ([raw file](https://github.com/google/mobly-bundled-snippets/blob/1e26712496c67c66a7871fb4e0ec1de197b741dc/src/main/java/com/google/android/mobly/snippet/bundled/AudioSnippet.java#L28)):*
<details><summary><i>Previously, adorokhine (Alexander Dorokhine) wrote…</i></summary><blockquote>

should be named mAudioManager per the style guide
</blockquote></details>

Done.

---

*[src/main/java/com/google/android/mobly/snippet/bundled/AudioSnippet.java, line 34 at r5](https://reviewable.io:443/reviews/google/mobly-bundled-snippets/14#-Kcu4H67VP_RAJTw8xwZ:-KcuxlTkju_doMH1id6t:b-896fix) ([raw file](https://github.com/google/mobly-bundled-snippets/blob/1e26712496c67c66a7871fb4e0ec1de197b741dc/src/main/java/com/google/android/mobly/snippet/bundled/AudioSnippet.java#L34)):*
<details><summary><i>Previously, adorokhine (Alexander Dorokhine) wrote…</i></summary><blockquote>

Indentation on empty line
</blockquote></details>

Done.

---

*[src/main/java/com/google/android/mobly/snippet/bundled/AudioSnippet.java, line 36 at r5](https://reviewable.io:443/reviews/google/mobly-bundled-snippets/14#-Kcu4PnMeP31jXLewsYZ:-Kcv6gFxbuqPbhs6Rx79:b-896fix) ([raw file](https://github.com/google/mobly-bundled-snippets/blob/1e26712496c67c66a7871fb4e0ec1de197b741dc/src/main/java/com/google/android/mobly/snippet/bundled/AudioSnippet.java#L36)):*
<details><summary><i>Previously, adorokhine (Alexander Dorokhine) wrote…</i></summary><blockquote>

If it returns the STREAM_MUSIC volume, maybe it should be called getMusicVolume?
</blockquote></details>

Done.

---

*[src/main/java/com/google/android/mobly/snippet/bundled/AudioSnippet.java, line 36 at r5](https://reviewable.io:443/reviews/google/mobly-bundled-snippets/14#-Kcu4KymRVxMRspEJ9jW:-KcvFf3Gp5CwlNkUzl8W:bu5urxg) ([raw file](https://github.com/google/mobly-bundled-snippets/blob/1e26712496c67c66a7871fb4e0ec1de197b741dc/src/main/java/com/google/android/mobly/snippet/bundled/AudioSnippet.java#L36)):*
<details><summary><i>Previously, adorokhine (Alexander Dorokhine) wrote…</i></summary><blockquote>

Does the snippet lib handle the primitive int correctly?
</blockquote></details>

As a return value, yes.

---

*[src/main/java/com/google/android/mobly/snippet/bundled/AudioSnippet.java, line 42 at r5](https://reviewable.io:443/reviews/google/mobly-bundled-snippets/14#-Kcu4UsfRqmQ_Embm8lT:-KcvARbSK98Et1oIQqdp:b-896fix) ([raw file](https://github.com/google/mobly-bundled-snippets/blob/1e26712496c67c66a7871fb4e0ec1de197b741dc/src/main/java/com/google/android/mobly/snippet/bundled/AudioSnippet.java#L42)):*
<details><summary><i>Previously, adorokhine (Alexander Dorokhine) wrote…</i></summary><blockquote>

What does 0 mean? Callsite comments?
</blockquote></details>

Done.

---

*[src/main/java/com/google/android/mobly/snippet/bundled/AudioSnippet.java, line 63 at r5](https://reviewable.io:443/reviews/google/mobly-bundled-snippets/14#-Kcu51bzxUKv9GEjtgdO:-KcvADW5qkttIzx4xAhi:b-263srn) ([raw file](https://github.com/google/mobly-bundled-snippets/blob/1e26712496c67c66a7871fb4e0ec1de197b741dc/src/main/java/com/google/android/mobly/snippet/bundled/AudioSnippet.java#L63)):*
<details><summary><i>Previously, adorokhine (Alexander Dorokhine) wrote…</i></summary><blockquote>

Rather than having to maintain this list, can we just iterate over AudioSystem.getNumStreamTypes() and mute each one?
</blockquote></details>

That seems to be a hidden API despite being referenced in public documentation.. I'm looking to see if there's a public getter.
In other news, value changed to 0 for all of them.

---

*[src/main/java/com/google/android/mobly/snippet/bundled/AudioSnippet.java, line 66 at r5](https://reviewable.io:443/reviews/google/mobly-bundled-snippets/14#-Kcu5Aeop_J2iRjZiJGT:-KcvGM9Bd8CABrvzxLvU:b-i576xn) ([raw file](https://github.com/google/mobly-bundled-snippets/blob/1e26712496c67c66a7871fb4e0ec1de197b741dc/src/main/java/com/google/android/mobly/snippet/bundled/AudioSnippet.java#L66)):*
<details><summary><i>Previously, adorokhine (Alexander Dorokhine) wrote…</i></summary><blockquote>

Does this mean vibration?
</blockquote></details>

Yes, at least for N.  

---

*[src/main/java/com/google/android/mobly/snippet/bundled/AudioSnippet.java, line 76 at r5](https://reviewable.io:443/reviews/google/mobly-bundled-snippets/14#-Kcu5D3moYEK-oSmx2ZQ:-KcvAbKW5tmTOGQUeKJe:b-896fix) ([raw file](https://github.com/google/mobly-bundled-snippets/blob/1e26712496c67c66a7871fb4e0ec1de197b741dc/src/main/java/com/google/android/mobly/snippet/bundled/AudioSnippet.java#L76)):*
<details><summary><i>Previously, adorokhine (Alexander Dorokhine) wrote…</i></summary><blockquote>

Alignment/indentation issues
</blockquote></details>

Done.

---

*[src/main/java/com/google/android/mobly/snippet/bundled/TelephonySnippet.java, line 52 at r5](https://reviewable.io:443/reviews/google/mobly-bundled-snippets/14#-Kcu5dNKZVITVuwGNC4i:-KcvBzuIt3_UuzprOl9G:b-896fix) ([raw file](https://github.com/google/mobly-bundled-snippets/blob/1e26712496c67c66a7871fb4e0ec1de197b741dc/src/main/java/com/google/android/mobly/snippet/bundled/TelephonySnippet.java#L52)):*
<details><summary><i>Previously, adorokhine (Alexander Dorokhine) wrote…</i></summary><blockquote>

Needs to mention what this int represents.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/google/mobly-bundled-snippets/14)*
<!-- Sent from Reviewable.io -->

dthkao commented:
Review status: 1 of 3 files reviewed at latest revision, 18 unresolved discussions, some commit checks failed.

---

*[src/main/java/com/google/android/mobly/snippet/bundled/TelephonySnippet.java, line 48 at r3](https://reviewable.io:443/reviews/google/mobly-bundled-snippets/14#-Kcu3xDmqKYAsSQMyBnk-r3-48:-KcvHjM5qvyIFnEDsj5m:b-q0cjig) ([raw file](https://github.com/google/mobly-bundled-snippets/blob/61771354104ea110721b2d83a8c14d78f339e329/src/main/java/com/google/android/mobly/snippet/bundled/TelephonySnippet.java#L48)):*
<details><summary><i>Previously, xpconanfan (Ang Li) wrote…</i></summary><blockquote>

What's in CellInfo?
Does Gson do the job or do we need to add a specific method for it in JsonBuilder?
</blockquote></details>

See response to Alex.

---

*[src/main/java/com/google/android/mobly/snippet/bundled/TelephonySnippet.java, line 28 at r5](https://reviewable.io:443/reviews/google/mobly-bundled-snippets/14#-Kcu5Gk4trZnob3aTWjf:-KcvHgCDerTvG2BNuycH:b-896fix) ([raw file](https://github.com/google/mobly-bundled-snippets/blob/1e26712496c67c66a7871fb4e0ec1de197b741dc/src/main/java/com/google/android/mobly/snippet/bundled/TelephonySnippet.java#L28)):*
<details><summary><i>Previously, adorokhine (Alexander Dorokhine) wrote…</i></summary><blockquote>

mTelephonyManager
</blockquote></details>

Done.

---

*[src/main/java/com/google/android/mobly/snippet/bundled/TelephonySnippet.java, line 47 at r5](https://reviewable.io:443/reviews/google/mobly-bundled-snippets/14#-Kcu5RHoRYm8qfIMLY68:-KcvHl4fns45BZrb8Y9t:b-mydnbs) ([raw file](https://github.com/google/mobly-bundled-snippets/blob/1e26712496c67c66a7871fb4e0ec1de197b741dc/src/main/java/com/google/android/mobly/snippet/bundled/TelephonySnippet.java#L47)):*
It works as it did in sl4a -- it seems like it just invokes toString() for each cellinfo:
>>> pprint(s.getAllCellInfo())
[u'CellInfoLte:{mRegistered=YES mTimeStampType=oem_ril mTimeStamp=240321865182133ns CellIdentityLte:{ mMcc=310 mMnc=260 mCi=21404418 mPci=94 mTac=14410 mEarfcn=2300} CellSignalStrengthLte: ss=31 rsrp=-86 rsrq=-14 rssnr=2147483647 cqi=2147483647 ta=2}',
 u'CellInfoLte:{mRegistered=NO mTimeStampType=oem_ril mTimeStamp=240321865182133ns CellIdentityLte:{ mMcc=2147483647 mMnc=2147483647 mCi=2147483647 mPci=360 mTac=2147483647 mEarfcn=2300} CellSignalStrengthLte: ss=26 rsrp=-90 rsrq=-13 rssnr=2147483647 cqi=2147483647 ta=2147483647}']

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/google/mobly-bundled-snippets/14)*
<!-- Sent from Reviewable.io -->

xpconanfan commented:
Review status: 1 of 3 files reviewed at latest revision, 18 unresolved discussions, some commit checks failed.

---

*[src/main/java/com/google/android/mobly/snippet/bundled/TelephonySnippet.java, line 47 at r5](https://reviewable.io:443/reviews/google/mobly-bundled-snippets/14#-Kcu5RHoRYm8qfIMLY68:-KcvYBE87AZOQYXf9sm5:b-4kj8hi) ([raw file](https://github.com/google/mobly-bundled-snippets/blob/1e26712496c67c66a7871fb4e0ec1de197b741dc/src/main/java/com/google/android/mobly/snippet/bundled/TelephonySnippet.java#L47)):*
<details><summary><i>Previously, dthkao (David T.H. Kao) wrote…</i></summary><blockquote>

It works as it did in sl4a -- it seems like it just invokes toString() for each cellinfo:
>>> pprint(s.getAllCellInfo())
[u'CellInfoLte:{mRegistered=YES mTimeStampType=oem_ril mTimeStamp=240321865182133ns CellIdentityLte:{ mMcc=310 mMnc=260 mCi=21404418 mPci=94 mTac=14410 mEarfcn=2300} CellSignalStrengthLte: ss=31 rsrp=-86 rsrq=-14 rssnr=2147483647 cqi=2147483647 ta=2}',
 u'CellInfoLte:{mRegistered=NO mTimeStampType=oem_ril mTimeStamp=240321865182133ns CellIdentityLte:{ mMcc=2147483647 mMnc=2147483647 mCi=2147483647 mPci=360 mTac=2147483647 mEarfcn=2300} CellSignalStrengthLte: ss=26 rsrp=-90 rsrq=-13 rssnr=2147483647 cqi=2147483647 ta=2147483647}']


</blockquote></details>

We don't want it to be strings though. We want proper json?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/google/mobly-bundled-snippets/14)*
<!-- Sent from Reviewable.io -->

dthkao commented:
*[src/main/java/com/google/android/mobly/snippet/bundled/TelephonySnippet.java, line 47 at r5](https://reviewable.io:443/reviews/google/mobly-bundled-snippets/14#-Kcu5RHoRYm8qfIMLY68:-Kcvi2Q77FHJQWIxgv2Z:b8sfme6) ([raw file](https://github.com/google/mobly-bundled-snippets/blob/1e26712496c67c66a7871fb4e0ec1de197b741dc/src/main/java/com/google/android/mobly/snippet/bundled/TelephonySnippet.java#L47)):*
<details><summary><i>Previously, xpconanfan (Ang Li) wrote…</i></summary><blockquote>

We don't want it to be strings though. We want proper json?
</blockquote></details>

I think that's outside the scope of this PR. We shouldn't be doing one-off conversion to JSON objects for every complex return type. That'd be either a change in the json serializer in MBS or result handling in snippet lib itself.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/google/mobly-bundled-snippets/14)*
<!-- Sent from Reviewable.io -->

adorokhine commented:
Reviewed 1 of 3 files at r6.
Review status: 1 of 3 files reviewed at latest revision, 14 unresolved discussions.

---

*[src/main/java/com/google/android/mobly/snippet/bundled/AudioSnippet.java, line 63 at r5](https://reviewable.io:443/reviews/google/mobly-bundled-snippets/14#-Kcu51bzxUKv9GEjtgdO:-KcyJfIKWUqT1Y5n0JVC:by4o6n) ([raw file](https://github.com/google/mobly-bundled-snippets/blob/1e26712496c67c66a7871fb4e0ec1de197b741dc/src/main/java/com/google/android/mobly/snippet/bundled/AudioSnippet.java#L63)):*
<details><summary><i>Previously, dthkao (David T.H. Kao) wrote…</i></summary><blockquote>

That seems to be a hidden API despite being referenced in public documentation.. I'm looking to see if there's a public getter.
In other news, value changed to 0 for all of them.
</blockquote></details>

Weird. Maybe we should file a bug?
Can we still use the deprecated AudioManager.NUM_STREAMS constant then?

---

*[src/main/java/com/google/android/mobly/snippet/bundled/AudioSnippet.java, line 66 at r5](https://reviewable.io:443/reviews/google/mobly-bundled-snippets/14#-Kcu5Aeop_J2iRjZiJGT:-KcyJjsUyPTRCUhHyINy:b-45ngpg) ([raw file](https://github.com/google/mobly-bundled-snippets/blob/1e26712496c67c66a7871fb4e0ec1de197b741dc/src/main/java/com/google/android/mobly/snippet/bundled/AudioSnippet.java#L66)):*
<details><summary><i>Previously, dthkao (David T.H. Kao) wrote…</i></summary><blockquote>

Yes, at least for N.  
</blockquote></details>

Can we clarify that in the comment?

---

*[src/main/java/com/google/android/mobly/snippet/bundled/AudioSnippet.java, line 32 at r6](https://reviewable.io:443/reviews/google/mobly-bundled-snippets/14#-KcyIRHPZjXcVXi6SrNm:-KcyIRHPZjXcVXi6SrNn:b-2x0kxp) ([raw file](https://github.com/google/mobly-bundled-snippets/blob/77c0051d893e60be598e451e63cf5b061458bc2c/src/main/java/com/google/android/mobly/snippet/bundled/AudioSnippet.java#L32)):*
> ```Java
>     public AudioSnippet() {
>         Context context = InstrumentationRegistry.getContext();
>         this.mAudioManager = (AudioManager) context.getSystemService(Context.AUDIO_SERVICE);
> ```

'this' prefix is not needed

---

*[src/main/java/com/google/android/mobly/snippet/bundled/TelephonySnippet.java, line 32 at r6](https://reviewable.io:443/reviews/google/mobly-bundled-snippets/14#-KcyJnPdhEomLDGyXYaE:-KcyJnPdhEomLDGyXYaF:b54lbbv) ([raw file](https://github.com/google/mobly-bundled-snippets/blob/77c0051d893e60be598e451e63cf5b061458bc2c/src/main/java/com/google/android/mobly/snippet/bundled/TelephonySnippet.java#L32)):*
> ```Java
>     public TelephonySnippet() {
>         Context context = InstrumentationRegistry.getContext();
>         this.mTelephonyManager = (TelephonyManager) context.getSystemService(
> ```

'this' not needed

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/google/mobly-bundled-snippets/14)*
<!-- Sent from Reviewable.io -->

dthkao commented:
Review status: 1 of 3 files reviewed at latest revision, 14 unresolved discussions.

---

*[src/main/java/com/google/android/mobly/snippet/bundled/AudioSnippet.java, line 63 at r5](https://reviewable.io:443/reviews/google/mobly-bundled-snippets/14#-Kcu51bzxUKv9GEjtgdO:-KcyMBX7bdr8qJIyA21j:b-wqu7ag) ([raw file](https://github.com/google/mobly-bundled-snippets/blob/1e26712496c67c66a7871fb4e0ec1de197b741dc/src/main/java/com/google/android/mobly/snippet/bundled/AudioSnippet.java#L63)):*
<details><summary><i>Previously, adorokhine (Alexander Dorokhine) wrote…</i></summary><blockquote>

Weird. Maybe we should file a bug?
Can we still use the deprecated AudioManager.NUM_STREAMS constant then?
</blockquote></details>

Deprecated one seems to work (AFAICT).

---

*[src/main/java/com/google/android/mobly/snippet/bundled/AudioSnippet.java, line 66 at r5](https://reviewable.io:443/reviews/google/mobly-bundled-snippets/14#-Kcu5Aeop_J2iRjZiJGT:-KcyMbleQCG_jpbImRmE:b-896fix) ([raw file](https://github.com/google/mobly-bundled-snippets/blob/1e26712496c67c66a7871fb4e0ec1de197b741dc/src/main/java/com/google/android/mobly/snippet/bundled/AudioSnippet.java#L66)):*
<details><summary><i>Previously, adorokhine (Alexander Dorokhine) wrote…</i></summary><blockquote>

Can we clarify that in the comment?
</blockquote></details>

Done.

---

*[src/main/java/com/google/android/mobly/snippet/bundled/AudioSnippet.java, line 32 at r6](https://reviewable.io:443/reviews/google/mobly-bundled-snippets/14#-KcyIRHPZjXcVXi6SrNm:-KcyM7oM-WP2vWctrk34:b-896fix) ([raw file](https://github.com/google/mobly-bundled-snippets/blob/77c0051d893e60be598e451e63cf5b061458bc2c/src/main/java/com/google/android/mobly/snippet/bundled/AudioSnippet.java#L32)):*
<details><summary><i>Previously, adorokhine (Alexander Dorokhine) wrote…</i></summary><blockquote>

'this' prefix is not needed
</blockquote></details>

Done.

---

*[src/main/java/com/google/android/mobly/snippet/bundled/TelephonySnippet.java, line 32 at r6](https://reviewable.io:443/reviews/google/mobly-bundled-snippets/14#-KcyJnPdhEomLDGyXYaE:-KcyMci0zKa3Bf33aB-M:b-896fix) ([raw file](https://github.com/google/mobly-bundled-snippets/blob/77c0051d893e60be598e451e63cf5b061458bc2c/src/main/java/com/google/android/mobly/snippet/bundled/TelephonySnippet.java#L32)):*
<details><summary><i>Previously, adorokhine (Alexander Dorokhine) wrote…</i></summary><blockquote>

'this' not needed
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/google/mobly-bundled-snippets/14)*
<!-- Sent from Reviewable.io -->

adorokhine commented:
Reviewed 2 of 3 files at r6.
Review status: 1 of 3 files reviewed at latest revision, 9 unresolved discussions.

---

*[src/main/java/com/google/android/mobly/snippet/bundled/AudioSnippet.java, line 57 at r7](https://reviewable.io:443/reviews/google/mobly-bundled-snippets/14#-KcyfLrenNnEmq7gjU9A:-KcyfLrenNnEmq7gjU9B:bb8nmvt) ([raw file](https://github.com/google/mobly-bundled-snippets/blob/9d222358794e24299062fd777aa6b70ab8ab9673/src/main/java/com/google/android/mobly/snippet/bundled/AudioSnippet.java#L57)):*
> ```Java
>     @Rpc(description = "Silences all audio streams.")
>     public void muteAll() {
>         for (int i=0; i<AudioManager.NUM_STREAMS; i++) {
> ```

Not too familiar with the AOSP style guide (Ang to confirm?) but usually there's spacing around operators like = and <:
`for (int i = 0; i < AudioManager.NUM_STREAMS; i++)`

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/google/mobly-bundled-snippets/14)*
<!-- Sent from Reviewable.io -->

dthkao commented:
Review status: 1 of 3 files reviewed at latest revision, 9 unresolved discussions.

---

*[src/main/java/com/google/android/mobly/snippet/bundled/AudioSnippet.java, line 57 at r7](https://reviewable.io:443/reviews/google/mobly-bundled-snippets/14#-KcyfLrenNnEmq7gjU9A:-KcyhhzqLEslYKZdqRZR:bsx592b) ([raw file](https://github.com/google/mobly-bundled-snippets/blob/9d222358794e24299062fd777aa6b70ab8ab9673/src/main/java/com/google/android/mobly/snippet/bundled/AudioSnippet.java#L57)):*
<details><summary><i>Previously, adorokhine (Alexander Dorokhine) wrote…</i></summary><blockquote>

Not too familiar with the AOSP style guide (Ang to confirm?) but usually there's spacing around operators like = and <:
`for (int i = 0; i < AudioManager.NUM_STREAMS; i++)`
</blockquote></details>

Can't find an explicit rule but their examples all are spaced, so fixed this.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/google/mobly-bundled-snippets/14)*
<!-- Sent from Reviewable.io -->

adorokhine commented:
Reviewed 2 of 2 files at r7, 2 of 2 files at r8.
Review status: all files reviewed at latest revision, 8 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/google/mobly-bundled-snippets/14)*
<!-- Sent from Reviewable.io -->

xpconanfan commented:
Review status: all files reviewed at latest revision, 9 unresolved discussions.

---

*[src/main/java/com/google/android/mobly/snippet/bundled/AudioSnippet.java, line 57 at r7](https://reviewable.io:443/reviews/google/mobly-bundled-snippets/14#-KcyfLrenNnEmq7gjU9A:-Kcyq9k9Sb76vM8wEBuN:b-14kbh2) ([raw file](https://github.com/google/mobly-bundled-snippets/blob/9d222358794e24299062fd777aa6b70ab8ab9673/src/main/java/com/google/android/mobly/snippet/bundled/AudioSnippet.java#L57)):*
<details><summary><i>Previously, dthkao (David T.H. Kao) wrote…</i></summary><blockquote>

Can't find an explicit rule but their examples all are spaced, so fixed this.
</blockquote></details>

Can you run the linter?
./gradlew goJF

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/google/mobly-bundled-snippets/14)*
<!-- Sent from Reviewable.io -->

dthkao commented:
Review status: 1 of 4 files reviewed at latest revision, 9 unresolved discussions.

---

*[src/main/java/com/google/android/mobly/snippet/bundled/AudioSnippet.java, line 57 at r7](https://reviewable.io:443/reviews/google/mobly-bundled-snippets/14#-KcyfLrenNnEmq7gjU9A:-Kcz-mdCD4g8orvhV9EA:b-896fix) ([raw file](https://github.com/google/mobly-bundled-snippets/blob/9d222358794e24299062fd777aa6b70ab8ab9673/src/main/java/com/google/android/mobly/snippet/bundled/AudioSnippet.java#L57)):*
<details><summary><i>Previously, xpconanfan (Ang Li) wrote…</i></summary><blockquote>

Can you run the linter?
./gradlew goJF
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/google/mobly-bundled-snippets/14)*
<!-- Sent from Reviewable.io -->

xpconanfan commented on src/main/java/com/google/android/mobly/snippet/bundled/AudioSnippet.java:
```diff
@@ -0,0 +1,79 @@
+    }
+
+    @Rpc(description = "Sets the music stream volume.")
+    public void setMusicVolume(Integer value) {
```
can we include the range of the input value in description?

xpconanfan commented on src/main/java/com/google/android/mobly/snippet/bundled/AudioSnippet.java:
```diff
@@ -0,0 +1,79 @@
+    }
+
+    @Rpc(description = "Sets the ringer volume.")
+    public void setRingVolume(Integer value) {
```
can we include the range of the input value in description?

xpconanfan commented on src/main/java/com/google/android/mobly/snippet/bundled/AudioSnippet.java:
```diff
@@ -0,0 +1,79 @@
+
+    @Rpc(description = "Silences all audio streams.")
+    public void muteAll() {
+        for (int i = 0; i < AudioManager.NUM_STREAMS; i++) {
```
`AudioManager.NUM_STREAMS` seems deprecated

> This constant was deprecated in API level 3. Use AudioSystem.getNumStreamTypes() instead

dthkao commented:
Review status: 1 of 4 files reviewed at latest revision, 12 unresolved discussions.

---

*[src/main/java/com/google/android/mobly/snippet/bundled/AudioSnippet.java, line 41 at r9](https://reviewable.io:443/reviews/google/mobly-bundled-snippets/14#-Kcu3xDmqKYAsSQMyBnj-r9-41:-Kd2FjCgw0upnrRjvK6k:b-896fix) ([raw file](https://github.com/google/mobly-bundled-snippets/blob/cf4b638c5d32646b00270881208cd3deca58d9f4/src/main/java/com/google/android/mobly/snippet/bundled/AudioSnippet.java#L41)):*
<details><summary><i>Previously, xpconanfan (Ang Li) wrote…</i></summary><blockquote>

can we include the range of the input value in description?
</blockquote></details>

Done.

---

*[src/main/java/com/google/android/mobly/snippet/bundled/AudioSnippet.java, line 52 at r9](https://reviewable.io:443/reviews/google/mobly-bundled-snippets/14#-Kcu3xDmqKYAsSQMyBnj-r9-52:-Kd2Fi5KnzD6t8uEr_G6:b-896fix) ([raw file](https://github.com/google/mobly-bundled-snippets/blob/cf4b638c5d32646b00270881208cd3deca58d9f4/src/main/java/com/google/android/mobly/snippet/bundled/AudioSnippet.java#L52)):*
<details><summary><i>Previously, xpconanfan (Ang Li) wrote…</i></summary><blockquote>

can we include the range of the input value in description?
</blockquote></details>

Done.

---

*[src/main/java/com/google/android/mobly/snippet/bundled/AudioSnippet.java, line 58 at r9](https://reviewable.io:443/reviews/google/mobly-bundled-snippets/14#-Kcu3xDmqKYAsSQMyBnj-r9-58:-Kd2FyBXYveCCNAqu4Pf:bwhycp4) ([raw file](https://github.com/google/mobly-bundled-snippets/blob/cf4b638c5d32646b00270881208cd3deca58d9f4/src/main/java/com/google/android/mobly/snippet/bundled/AudioSnippet.java#L58)):*
<details><summary><i>Previously, xpconanfan (Ang Li) wrote…</i></summary><blockquote>

`AudioManager.NUM_STREAMS` seems deprecated

> This constant was deprecated in API level 3. Use AudioSystem.getNumStreamTypes() instead


</blockquote></details>

Documentation may have an issue. See previous comments.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/google/mobly-bundled-snippets/14)*
<!-- Sent from Reviewable.io -->
